internal/wasi/wasi.go

```diff
@@ -1296,19 +1313,13 @@ func (a *wasiAPI) FdTell(ctx wasm.Module, fd, resultOffset uint32) wasi.Errno {
 func (a *wasiAPI) FdWrite(ctx wasm.Module, fd, iovs, iovsCount, resultSize uint32) wasi.Errno {
 	cfg := a.config(ctx.Context())
 
-	var writer io.Writer
-
-	switch fd {
-	case 1:
-		writer = cfg.stdout
-	case 2:
-		writer = cfg.stderr
-	default:
-		f, ok := cfg.opened[fd]
-		if !ok || f.file == nil {
-			return wasi.ErrnoBadf
-		}
-		writer = f.file
+	f, ok := cfg.opened[fd]
+	if !ok || f.file == nil {
```

Comment:
comment on what `f.file == nil` means?

Reply:
It was when `f` was a pre-opened directory, but I've changed the logic so that `f.file` never becomes `nil`.